cpp/include/cudf/aggregation.hpp

```diff
@@ -253,6 +255,44 @@ std::unique_ptr<Base> make_nth_element_aggregation(
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_row_number_aggregation();
 
+/**
+ * @brief Factory to create a RANK aggregation
+ *
+ * `RANK` returns a non-nullable column of size_type ranks, the number of rows preceding
+ * or equal to the current row plus one. As a result, ranks are not unique and gaps will
+ * appear in the ranking sequence. `RANK` aggregations are not compatible with exclusive
+ * scans and will return a fully valid column regardless of null_handling policy specified.
+ *
+ * @code{.pseudo}
+ * order_by              = {{0, 0, 0, 1, 1, 1}, {0, 0, 1, 0, 1, 1}}
+ * ungrouped_scan_result = {1, 1, 3, 4, 5, 5}
```

Comment:
I don't understand this example. Is the order_by column a list column?

Reply:
Used more complete examples with better explanations -- modeled after some of the `rolling.hpp` documentation.